test/test_prepared.py

```diff
@@ -0,0 +1,279 @@
+import random
+import time
+
+import psycopg
+import pytest
+from psycopg import pq, sql
+
+from .utils import LIBPQ_SUPPORTS_PIPELINING, LINUX, USE_SUDO
+
+
+def test_prepared_statement(bouncer):
```

Comment:
This is a list of the manual testing that I know of that has been done, if you do/did any other please comment below:
- [x] I ran the `libpq_pipeline` binary from the postgres repo against pgbouncer and no issues were found
- [x] @knizhnik ran the test suite of https://github.com/lib/pq and didn't find any issues related to protocol level prepared statements (only stuff for temporary tables, gucs and other session state)
- [ ] Npgsql
- [ ] JDBC